### PR TITLE
[code-infra] Tentative fix for Argos flaky screenshot tests

### PR DESCRIPTION
**Pull request by JCQuintas**
- Move to use TS instead of JS
- Add `page.waitForLoadState()` before screenshot

### CODE REVIEW

File: test/regressions/index.test.ts [hunk 1]
```diff
@@ -62,7 +62,7 @@ async function main() {
 
   // Wait for all requests to finish.
   // This should load shared resources such as fonts.
-  await page.goto(`${baseUrl}#no-dev`, { waitUntil: 'networkidle0' });
+  await page.goto(`${baseUrl}#no-dev`, { waitUntil: 'networkidle' });
```
Comment (JCQuintas) [inline test/regressions/index.test.ts:65]: This seemed a bug

File: test/regressions/index.test.ts [hunk 2]
```diff
@@ -160,6 +155,14 @@ async function main() {
           { timeout: 1000 },
         );
 
+        if (/^\docs-charts-.*/.test(pathURL)) {
+          // Run one tick of the clock to get the final animation state
+          await sleep(10);
+        }
+
+        // Wait for the page to settle after taking the screenshot.
+        await page.waitForLoadState();
```
Comment (JCQuintas) [inline test/regressions/index.test.ts:164]: new `waitForLoadState`

File: test/regressions/index.test.ts (hunk 1 above)
Comment (LukasTy) [reply test/regressions/index.test.ts:65]: Are you sure? 🤔 
https://www.webshare.io/academy-article/puppeteer-networkidle0-vs-networkidle2
it seems like it is correct. 🤔

File: test/regressions/.mocharc.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 module.exports = {
+  extension: ['js', 'ts', 'tsx'],
```
Comment (LukasTy) [inline test/regressions/.mocharc.js:2]: Do we need to duplicate it if this entry is present in `setupBabel`? 🤔

**Review by LukasTy — COMMENTED** [review]
Nice improvement, thanks for looking into this! 🙏 👍

File: test/regressions/index.test.ts (hunk 1 above)
Comment (JCQuintas) [reply test/regressions/index.test.ts:65]: We are using playwright, the types flag that as a type error

File: test/regressions/.mocharc.js (hunk 3 above)
Comment (JCQuintas) [reply test/regressions/.mocharc.js:2]: yeah, seems to throw a bunch of errors otherwise 🙃

Comment (LukasTy) [reply test/regressions/.mocharc.js:2]: Gotcha. 👌

File: test/regressions/index.test.ts (hunk 1 above)
Comment (LukasTy) [reply test/regressions/index.test.ts:65]: Ok, nice. 👍 
It's a difference between Playwright and Puppeteer. 🙈 
https://playwright.dev/docs/api/class-page#page-goto-option-wait-until

**Review by LukasTy — APPROVED** [review]
Let's test this. 🚀 🤞

**Comment by LukasTy** [comment]
WDYT, @JCQuintas, should we cherry-pick this to v7.x as we'll be doing a lot of PRs for that branch? 🤔

**Comment by JCQuintas** [comment]
probably, yeah

**Pull request merged**